README.md

```diff
@@ -276,55 +276,80 @@ An AppArmor profile is available for mandatory access control. When installing s
+We use `qtbot`, bundled with the [pytest-qt](https://pytest-qt.readthedocs.io/en/latest/) package, for UI interaction within our functional tests.
 
 #### Generating new cassettes
 
-Some changes may require generating new cassettes, such as modifications to API endpoints. If you see the following warning, you may need to generate a new cassette or all new cassettes, depending on the change you made:
+We use [vcrpy](https://vcrpy.readthedocs.io/en/latest/) to record and replay API calls. Each request made from a test and response from the server is stored in a "cassette" yaml file in the `tests/functional/cassettes` directory.
+
+If the SDK changes its API, then you'll see the following warning indicating that a request failed to be found in an existing cassette and that you'll need to regenerate cassettes:
 
 ```
 Can't overwrite existing cassette ('<path-to-cassette-for-a-functional-test>') in your current record mode ('once').
 ```
 
-To generate new cassettes, follow these instructions:
-
-1. Before generating new cassettes that will be used to replay server interaction, bypass TOTP verification so that we can use the hard-coded value of `123456` in `tests/conftest.py`. You can do this by applying the following diff to the securedrop server, which you will be running in the next step:
-       ```diff
-       diff --git a/securedrop/models.py b/securedrop/models.py
-       index dcd26bbaf..50bc491b4 100644
-       --- a/securedrop/models.py
-       +++ b/securedrop/models.py
-       @@ -649,6 +649,7 @@ class Journalist(db.Model):
-
-                try:
-                    user = Journalist.query.filter_by(username=username).one()
-       +            return user
-                except NoResultFound:
-                    raise InvalidUsernameException(
-                        "invalid username '{}'".format(username))
-       (END)
-       ```
-2. Start the dev server: `NUM_SOURCES=0 make dev`
-3. Set up the dev server with data required for functional tests to pass and run in any order:
-    - Create two new sources, each with one submission that contains both a file and a message. The message should be set to `this is the message`. The file should be called `hello.txt` and contain a single line of text: `hello`.
-3. Delete all the old cassettes by running `rm -r tests/functional/cassettes` or just delete the cassettes you wish to regenerate.
-4. Run `make test-functional` which will regenerate cassettes for any that are missing in the `tests/functional/cassettes` folder.
+To set up a local dev server and generate cassettes, follow these steps:
+
+1. Bypass TOTP verification so that we can use the TOTP value of `123456` hard-coded in `tests/conftest.py`. You can do this by applying the following patch to the server code:
+
+https://gist.github.com/creviera/8793d5ec4d28f034f2c1e8320a93866a
```

Comment:
This is a good idea, definitely more readable and easier to review/use, provided the author does not delete it